src/i3_json_parser/i3_json_parser-utility.cpp

```diff
@@ -197,6 +197,34 @@ namespace
         const auto val = get_attribute_value<std::uint8_t>(a_json_object, "fullscreen_mode");
         return static_cast<i3_containers::fullscreen_mode_type>(val);
     }
+
+
+    /**
+     * \brief                      Extracts the marks of the node from the JSON object.
+     *
+     * \param [in] a_json_object   JSON object.
+     *
+     * \return                     Extracted vector of marks
+     */
+    std::vector<std::string> extract_marks(const rapidjson::Value& a_json_object)
+    {
+        assert(a_json_object.IsObject());
+        if (!a_json_object.HasMember("marks") || !a_json_object["marks"].IsArray())
+        {
+            return { };
+        }
```

Comment:
I'm curious about [this one](https://github.com/Iskustvo/i3-ipcpp/pull/8#discussion_r1590185309).
Did you find some case where `i3` doesn't provide this field or just wanted to be on the safe side?

Just to be clear, I don't mind it at all.
I know I initially pushed all "guaranteed" (by `i3`) things into asserts, but that might've been bad idea :smile:

Reply:
It was just to be safe, since I don't actually know if `i3` will always provide the field. The pattern is the same of the `window_properties` field, but maybe you prefer to switch to an `assert` like the one in `fullscreen_mode` to eventually get an error in debug build, but avoid paying the cost of the check in release?

Reply:
Well, `i3` [claims](https://github.com/i3/i3/pull/3970) that they do send it each time (as I also observed with limited testing), but it's okay to be on the safe side as well :wink:
Was asking mostly to know if we need to open bug in `i3` if you found case where we enter this `if`.

Yeah, at the time of writing, I greedily put everything that was "guaranteed" in `assert` and leave only things that actually had cases of missing fields in runtime, but no need to blindly follow that decision.